### PR TITLE
Fix circular imports and refactor submodules

### PR DESCRIPTION
**Pull request by NeonDaniel**
Add unit tests of `locations` module
Refactor modules for clarity

### CODE REVIEW

File: ovos_config/base.py [hunk 1]
```diff
@@ -1,272 +1,3 @@
...
-            self.load_local(self.path)
-
-
-class MycroftUserConfig(LocalConf):
-    def __init__(self):
-        super().__init__(USER_CONFIG)
-
-
-MycroftXDGConfig = MycroftUserConfig
+LOG.warning("This reference is deprecated, import from ovos_config.models")
```
Comment (JarbasAl) [inline ovos_config/base.py:3]: This file could be removed, it was added in previous pr only and that is a buggy release to be yanked

File: ovos_config/__init__.py [hunk 2]
```diff
@@ -3,4 +3,4 @@
     set_default_tz, set_default_lang, get_default_tz, get_default_lang, \
     get_config_tz, get_primary_lang_code, load_languages, load_language
 from ovos_config.locations import SYSTEM_CONFIG, USER_CONFIG, DEFAULT_CONFIG, get_xdg_config_locations
-from ovos_config.ovos import get_ovos_config
+from ovos_config.meta_config import get_ovos_config
```
Comment (JarbasAl) [inline ovos_config/__init__.py:6]: Can we name this just meta? Avoid repeating words in the import

File: ovos_config/base.py (hunk 1 above)
Comment (NeonDaniel) [reply ovos_config/base.py:3]: Ah, right. I'll just remove it entirely

File: ovos_config/__init__.py (hunk 2 above)
Comment (NeonDaniel) [reply ovos_config/__init__.py]: sure

File: ovos_config/ovos.py [hunk 3]
```diff
@@ -1,211 +1,4 @@
...
-
-    different derivative cores may change this file, this value is derived from ovos.conf
-
-    NOTE: this value will be set globally, per core overrides in ovos.conf take precedence
-    """
-    file_path = file_path or _oloc.find_default_config()
-    LOG.info(f"default config file changed to: {file_path}")
-    save_ovos_config({"default_config_path": file_path})
+LOG.warning("This reference is deprecated, "
+            "import from ovos_config.meta_config")
```
Comment (JarbasAl) [inline ovos_config/ovos.py:4]: missed the rename here

Comment (NeonDaniel) [reply ovos_config/ovos.py]: Fixed

**Pull request merged**